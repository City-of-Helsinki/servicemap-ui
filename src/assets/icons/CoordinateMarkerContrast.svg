<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<circle cx="12" cy="12" r="11.5" fill="#313131" stroke="white"/>
<mask id="mask0" mask-type="alpha" maskUnits="userSpaceOnUse" x="7" y="5" width="10" height="14">
<path fill-rule="evenodd" clip-rule="evenodd" d="M12 5.33333C9.42004 5.33333 7.33337 7.42 7.33337 10C7.33337 13.5 12 18.6667 12 18.6667C12 18.6667 16.6667 13.5 16.6667 10C16.6667 7.42 14.58 5.33333 12 5.33333ZM12 11.6667C11.08 11.6667 10.3334 10.92 10.3334 10C10.3334 9.08 11.08 8.33333 12 8.33333C12.92 8.33333 13.6667 9.08 13.6667 10C13.6667 10.92 12.92 11.6667 12 11.6667Z" fill="white"/>
</mask>
<g mask="url(#mask0)">
<rect x="-4.66663" y="-4.66667" width="33.3333" height="33.3333" fill="white"/>
<mask id="mask1" mask-type="alpha" maskUnits="userSpaceOnUse" x="-5" y="-5" width="34" height="34">
<rect x="-4.66663" y="-4.66667" width="33.3333" height="33.3333" fill="white"/>
</mask>
<g mask="url(#mask1)">
</g>
</g>
</svg>
