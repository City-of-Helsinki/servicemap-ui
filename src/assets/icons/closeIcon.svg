<svg width="18" height="18" viewBox="0 0 18 18" fill="none" xmlns="http://www.w3.org/2000/svg">
<mask id="mask0" mask-type="alpha" maskUnits="userSpaceOnUse" x="1" y="1" width="16" height="16">
<path fill-rule="evenodd" clip-rule="evenodd" d="M9 1.5C4.8525 1.5 1.5 4.8525 1.5 9C1.5 13.1475 4.8525 16.5 9 16.5C13.1475 16.5 16.5 13.1475 16.5 9C16.5 4.8525 13.1475 1.5 9 1.5ZM12.75 11.6925L11.6925 12.75L9 10.0575L6.3075 12.75L5.25 11.6925L7.9425 9L5.25 6.3075L6.3075 5.25L9 7.9425L11.6925 5.25L12.75 6.3075L10.0575 9L12.75 11.6925Z" fill="white"/>
</mask>
<g mask="url(#mask0)">
<rect x="-9.75" y="-9.75" width="37.5" height="37.5" fill="white"/>
<mask id="mask1" mask-type="alpha" maskUnits="userSpaceOnUse" x="-10" y="-10" width="38" height="38">
<rect x="-9.75" y="-9.75" width="37.5" height="37.5" fill="white"/>
</mask>
<g mask="url(#mask1)">
</g>
</g>
</svg>
