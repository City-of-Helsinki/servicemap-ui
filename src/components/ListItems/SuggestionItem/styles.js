export default theme => ({
  container: {
    display: 'flex',
    flex: '1 1 auto',
    padding: `${theme.spacing.unitHalf}px 0`,
  },
  listItem: {
    padding: `0 ${theme.spacing.unit}px`,
  },
  textContainer: {
    display: 'flex',
    padding: `${theme.spacing.unit}px 0`,
    marginLeft: theme.spacing.unitDouble,
    marginRight: theme.spacing.unitDouble,
    whiteSpace: 'pre-line',
    '& p': {
      lineHeight: '18px',
    },
  },
  listIcon: {
    width: '1.5rem',
    height: '1.5rem',
    margin: `${theme.spacing.unitHalf}px ${theme.spacing.unit}px`,
    marginRight: 0,
    alignSelf: 'center',
    padding: theme.spacing.unitHalf,
  },
  suggestIcon: {
    color: '#757575',
    height: 'auto',
    margin: 0,
    padding: 0,
  },
  suggestIconLabel: {
    padding: theme.spacing.unit,
  },
  subtitle: {
    fontSize: '10px',
    fontWeight: 'none',
    lineHeight: '18px',
  },
  divider: {
    marginLeft: theme.spacing.unit * 8,
    marginRight: -theme.spacing.unitDouble,
  },
  text: {
    alignSelf: 'center',
  },
});
