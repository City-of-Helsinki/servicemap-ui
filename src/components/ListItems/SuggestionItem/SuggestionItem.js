import React, { useState } from 'react';
import PropTypes from 'prop-types';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import {
  Typography, Divider, Button,
} from '@material-ui/core';
import { ArrowUpward } from '@material-ui/icons';
import { unstable_useMediaQuery as useMediaQuery } from '@material-ui/core/useMediaQuery';
import { intlShape } from 'react-intl';
import config from '../../../../config';
import BoldedText from '../../../utils/BoldedText';
import { keyboardHandler } from '../../../utils';

const SuggestionItem = (props) => {
  const {
    button,
    classes,
    divider,
    text,
    handleItemClick,
    handleArrowClick,
    icon,
    intl,
    link,
    selected,
    srText,
    subtitle,
    query,
  } = props;

  const [mouseDown, setMouseDown] = useState(false);
  const isMobile = useMediaQuery(`(max-width:${config.mobileUiBreakpoint}px)`);
  const onClick = (button || link) && handleItemClick
    ? (e) => {
      e.preventDefault();
      if (!mouseDown) {
        handleItemClick();
        setMouseDown(true);
      }
    } : null;

  return (
    <React.Fragment>
      <ListItem
        button={!!link || button}
        component="li"
        classes={{
          root: classes.listItem,
        }}
        selected={selected}
        className="suggestion"
        onClick={onClick}
        onMouseDown={onClick}
        onMouseUp={() => setMouseDown(false)}
        onKeyDown={() => {
          keyboardHandler(onClick, ['space', 'enter']);
        }}
        onKeyUp={() => setMouseDown(false)}
      >
        <span
          className={classes.container}
          type="submit"
          role="link"
          tabIndex="-1"
        >
          <ListItemIcon aria-hidden className={`${classes.listIcon} ${link ? classes.link : null}`}>
            {icon}
          </ListItemIcon>

          <ListItemText
            className={classes.text}
            classes={{ root: classes.textContainer }}
          >
            <Typography variant="srOnly">
              {`${srText || ''} ${text} ${subtitle || ''}`}
            </Typography>

            <Typography
              aria-hidden
              variant="body2"
              classes={{ root: link ? classes.link : null }}
            >
              {
                query
                  ? (
                    <BoldedText
                      text={text}
                      shouldBeBold={query}
                    />
                  ) : text
              }
            </Typography>
            {
            subtitle
            && (
              <Typography
                aria-hidden
                variant="body2"
                className={classes.subtitle}
              >
                {subtitle}
              </Typography>

            )
          }
          </ListItemText>
        </span>
        {
          isMobile
          && handleArrowClick
          && (
            <Button
              aria-label={intl.formatMessage({ id: 'search.arrowLabel' })}
              className={`${classes.suggestIcon} ${link ? classes.link : null}`}
              classes={{
                label: classes.suggestIconLabel,
              }}
              onMouseDown={(e) => {
                e.preventDefault();
                e.stopPropagation();
                const value = text.props ? text.props.text : text;
                handleArrowClick(value);
                return false;
              }}
            >
              <ArrowUpward style={{ transform: 'rotate(-48deg)' }} />
            </Button>
          )
        }
      </ListItem>
      {divider ? (
        <li>
          <Divider aria-hidden className={classes.divider} />
        </li>
      ) : null}
    </React.Fragment>
  );
};

export default SuggestionItem;

SuggestionItem.propTypes = {
  button: PropTypes.bool,
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  srText: PropTypes.string,
  intl: intlShape.isRequired,
  link: PropTypes.bool,
  icon: PropTypes.objectOf(PropTypes.any),
  handleArrowClick: PropTypes.func,
  handleItemClick: PropTypes.func,
  divider: PropTypes.bool,
  selected: PropTypes.bool,
  subtitle: PropTypes.string,
  query: PropTypes.string,
};

SuggestionItem.defaultProps = {
  button: false,
  srText: null,
  link: false,
  icon: null,
  handleArrowClick: null,
  handleItemClick: null,
  divider: false,
  selected: false,
  subtitle: null,
  query: null,
};
