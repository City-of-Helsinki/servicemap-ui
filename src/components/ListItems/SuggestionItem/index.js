import { withStyles } from '@mui/styles';
import { injectIntl } from 'react-intl';
import SuggestionItem from './SuggestionItem';
import styles from './styles';

export default injectIntl(withStyles(styles)(SuggestionItem));
