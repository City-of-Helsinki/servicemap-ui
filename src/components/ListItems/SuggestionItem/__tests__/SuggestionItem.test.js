// // Link.react.test.js
import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import { MuiThemeProvider } from '@material-ui/core';
import { IntlProvider } from 'react-intl';
import { Search } from '@material-ui/icons';
import themes from '../../../../themes';
import SuggestionItem from '../index';

// Mock props for intl provider
const intlMock = {
  locale: 'en',
  messages: {
    'search.arrowLabel': 'Arrow button label',
  },
  wrapRichTextChunksInFragment: false,
};

// Generic required props for SimpleListItem
const mockProps = {
  text: 'Title text',
  subtitle: 'Subtitle text',
  icon: <Search />,
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <IntlProvider {...intlMock}>
    <MuiThemeProvider theme={themes.SMTheme}>
      {children}
    </MuiThemeProvider>
  </IntlProvider>
);

describe('<SuggestionItem />', () => {
  // let render;
  let mount;

  beforeEach(() => {
    // render = createRender({ wrappingComponent: Providers });
    mount = createMount({ wrappingComponent: Providers });
  });

  afterEach(() => {
    mount.cleanUp();
  });

  const resetMount = () => {
    mount.cleanUp();
    mount = createMount({ wrappingComponent: Providers });
  };

  it('should work', () => {
    const component = mount(<SuggestionItem {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it('simulates click event', () => {
    const mockCallBack = jest.fn();
    const component = mount(<SuggestionItem {...mockProps} handleItemClick={mockCallBack} />);

    component.find('ForwardRef(ListItem)').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('simulates keyboard event', () => {
    const mockCallBack = jest.fn();
    const component = mount(<SuggestionItem {...mockProps} handleItemClick={mockCallBack} />);

    component.find('ForwardRef(ListItem)').simulate('keyDown', { key: 'enter', keyCode: 13, which: 13 });
    // Simulate keyUp to release SuggestionItem mouseDown state
    component.find('ForwardRef(ListItem)').simulate('keyUp', { key: 'enter', keyCode: 13, which: 13 });
    component.find('ForwardRef(ListItem)').simulate('keyDown', { key: 'space', keyCode: 32, which: 32 });
    expect(mockCallBack.mock.calls.length).toEqual(2);
  });

  it('does show text correctly', () => {
    const component = mount(<SuggestionItem {...mockProps} />);

    const p = component.find('p');
    expect(p.at(0).text()).toEqual(mockProps.text);
    expect(p.at(1).text()).toEqual(mockProps.subtitle);
  });

  it('does set select correctly', () => {
    let component = mount(<SuggestionItem {...mockProps} />);
    expect(component.find('ForwardRef(ListItem)').props().selected).toBeFalsy();

    resetMount();

    component = mount(<SuggestionItem {...mockProps} selected />);
    expect(component.find('ForwardRef(ListItem)').props().selected).toBeTruthy();
  });

  it('does set divider correctly', () => {
    let component = mount(<SuggestionItem {...mockProps} />);
    expect(component.find('ForwardRef(Divider)').exists()).toBeFalsy();

    resetMount();

    component = mount(<SuggestionItem {...mockProps} divider />);
    expect(component.find('ForwardRef(Divider)').exists()).toBeTruthy();
  });

  it('does bold query correctly', () => {
    const component = mount(<SuggestionItem {...mockProps} query="text" />);

    expect(component.find('b').text()).toEqual('text');
  });

  it('does use default accessibility attributes correctly', () => {
    const component = mount(<SuggestionItem {...mockProps} />);

    const srText = component.find('span').at(1);
    const text = component.find('p').at(0);
    const containsText = srText.text().indexOf(mockProps.text) !== -1
      && srText.text().indexOf(mockProps.subtitle) !== -1;

    // Expect screen reader texts to render correctly
    expect(containsText).toBeTruthy();
    // Expect aria-hidden attributes to be placed correctly
    expect(srText.props()['aria-hidden']).toBeFalsy();
    expect(text.props()['aria-hidden']).toBeTruthy();
    // Expect role to be set
    expect(component.find('ForwardRef(ListItem)').props().role).toEqual('link');
    // Expect element to have tabIndex 0
    expect(component.find('ForwardRef(ListItem)').props().tabIndex).toEqual('0');
    // Expect span element to have tabIndex -1
    expect(component.find('ForwardRef(ListItem)').find('span').at(0).props().tabIndex).toEqual('-1');
  });

  it('does use given accessibility attributes correctly', () => {
    const component = mount(
      <SuggestionItem
        {...mockProps}
        role="button"
        divider
      />,
    );

    // Expect role to be set
    expect(component.props().role).toEqual('button');
    // Expect divider element to be hidden from screen readers
    expect(component.find('li').at(1).props()['aria-hidden']).toBeTruthy();
    // Expect element to have tabIndex 0
    expect(component.find('ForwardRef(ListItem)').props().tabIndex).toEqual('0');
  });
});
