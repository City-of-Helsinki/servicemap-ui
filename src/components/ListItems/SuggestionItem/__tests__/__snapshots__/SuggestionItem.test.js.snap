// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuggestionItem /> should work 1`] = `
<injectIntl(WithStyles(SuggestionItem))
  icon={<pure(SearchIcon) />}
  subtitle="Subtitle text"
  text="Title text"
>
  <WithStyles(SuggestionItem)
    icon={<pure(SearchIcon) />}
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatDateToParts": [Function],
        "formatDisplayName": [Function],
        "formatList": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatNumberToParts": [Function],
        "formatPlural": [Function],
        "formatRelativeTime": [Function],
        "formatTime": [Function],
        "formatTimeToParts": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getDisplayNames": [Function],
          "getListFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralRules": [Function],
          "getRelativeTimeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "search.arrowLabel": "Arrow button label",
        },
        "onError": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": undefined,
        "wrapRichTextChunksInFragment": false,
      }
    }
    subtitle="Subtitle text"
    text="Title text"
  >
    <SuggestionItem
      classes={
        Object {
          "container": "SuggestionItem-container-1",
          "divider": "SuggestionItem-divider-8",
          "itemFocus": "SuggestionItem-itemFocus-10",
          "listIcon": "SuggestionItem-listIcon-4",
          "listItem": "SuggestionItem-listItem-2",
          "subtitle": "SuggestionItem-subtitle-7",
          "suggestIcon": "SuggestionItem-suggestIcon-5",
          "suggestIconLabel": "SuggestionItem-suggestIconLabel-6",
          "text": "SuggestionItem-text-9",
          "textContainer": "SuggestionItem-textContainer-3",
        }
      }
      divider={false}
      handleArrowClick={null}
      handleItemClick={null}
      icon={<pure(SearchIcon) />}
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatDateToParts": [Function],
          "formatDisplayName": [Function],
          "formatList": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatNumberToParts": [Function],
          "formatPlural": [Function],
          "formatRelativeTime": [Function],
          "formatTime": [Function],
          "formatTimeToParts": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getDisplayNames": [Function],
            "getListFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralRules": [Function],
            "getRelativeTimeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {
            "search.arrowLabel": "Arrow button label",
          },
          "onError": [Function],
          "textComponent": Symbol(react.fragment),
          "timeZone": undefined,
          "wrapRichTextChunksInFragment": false,
        }
      }
      query={null}
      selected={false}
      subtitle="Subtitle text"
      text="Title text"
    >
      <WithStyles(ForwardRef(ListItem))
        button={true}
        className="suggestion"
        classes={
          Object {
            "root": "SuggestionItem-listItem-2",
            "selected": "SuggestionItem-itemFocus-10",
          }
        }
        component="li"
        onClick={null}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={null}
        onMouseUp={[Function]}
        role="link"
        selected={false}
        tabIndex="0"
      >
        <ForwardRef(ListItem)
          button={true}
          className="suggestion"
          classes={
            Object {
              "alignItemsFlexStart": "MuiListItem-alignItemsFlexStart",
              "button": "MuiListItem-button",
              "container": "MuiListItem-container",
              "dense": "MuiListItem-dense",
              "disabled": "Mui-disabled",
              "divider": "MuiListItem-divider",
              "focusVisible": "Mui-focusVisible",
              "gutters": "MuiListItem-gutters",
              "root": "MuiListItem-root SuggestionItem-listItem-2",
              "secondaryAction": "MuiListItem-secondaryAction",
              "selected": "Mui-selected SuggestionItem-itemFocus-10",
            }
          }
          component="li"
          onClick={null}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={null}
          onMouseUp={[Function]}
          role="link"
          selected={false}
          tabIndex="0"
        >
          <WithStyles(ForwardRef(ButtonBase))
            className="MuiListItem-root SuggestionItem-listItem-2 suggestion MuiListItem-gutters MuiListItem-button"
            component="li"
            disabled={false}
            focusVisibleClassName="Mui-focusVisible"
            onClick={null}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={null}
            onMouseUp={[Function]}
            role="link"
            tabIndex="0"
          >
            <ForwardRef(ButtonBase)
              className="MuiListItem-root SuggestionItem-listItem-2 suggestion MuiListItem-gutters MuiListItem-button"
              classes={
                Object {
                  "disabled": "Mui-disabled",
                  "focusVisible": "Mui-focusVisible",
                  "root": "MuiButtonBase-root",
                }
              }
              component="li"
              disableRipple={true}
              disabled={false}
              focusVisibleClassName="Mui-focusVisible"
              onClick={null}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={null}
              onMouseUp={[Function]}
              role="link"
              tabIndex="0"
            >
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root SuggestionItem-listItem-2 suggestion MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onClick={null}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="link"
                tabIndex="0"
              >
                <span
                  className="SuggestionItem-container-1"
                  key=".0"
                  role="link"
                  tabIndex="-1"
                  type="submit"
                >
                  <WithStyles(ForwardRef(ListItemIcon))
                    aria-hidden={true}
                    className="SuggestionItem-listIcon-4"
                  >
                    <ForwardRef(ListItemIcon)
                      aria-hidden={true}
                      className="SuggestionItem-listIcon-4"
                      classes={
                        Object {
                          "alignItemsFlexStart": "MuiListItemIcon-alignItemsFlexStart",
                          "root": "MuiListItemIcon-root",
                        }
                      }
                    >
                      <div
                        aria-hidden={true}
                        className="MuiListItemIcon-root SuggestionItem-listIcon-4"
                      >
                        <pure(SearchIcon)>
                          <SearchIcon>
                            <WithStyles(ForwardRef(SvgIcon))>
                              <ForwardRef(SvgIcon)
                                classes={
                                  Object {
                                    "colorAction": "MuiSvgIcon-colorAction",
                                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                                    "colorError": "MuiSvgIcon-colorError",
                                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                    "root": "MuiSvgIcon-root",
                                  }
                                }
                              >
                                <svg
                                  aria-hidden={true}
                                  className="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                                  />
                                  <path
                                    d="M0 0h24v24H0z"
                                    fill="none"
                                  />
                                </svg>
                              </ForwardRef(SvgIcon)>
                            </WithStyles(ForwardRef(SvgIcon))>
                          </SearchIcon>
                        </pure(SearchIcon)>
                      </div>
                    </ForwardRef(ListItemIcon)>
                  </WithStyles(ForwardRef(ListItemIcon))>
                  <WithStyles(ForwardRef(ListItemText))
                    className="SuggestionItem-text-9"
                    classes={
                      Object {
                        "root": "SuggestionItem-textContainer-3",
                      }
                    }
                  >
                    <ForwardRef(ListItemText)
                      className="SuggestionItem-text-9"
                      classes={
                        Object {
                          "dense": "MuiListItemText-dense",
                          "inset": "MuiListItemText-inset",
                          "multiline": "MuiListItemText-multiline",
                          "primary": "MuiListItemText-primary",
                          "root": "MuiListItemText-root SuggestionItem-textContainer-3",
                          "secondary": "MuiListItemText-secondary",
                        }
                      }
                    >
                      <div
                        className="MuiListItemText-root SuggestionItem-textContainer-3 SuggestionItem-text-9"
                      >
                        <WithStyles(ForwardRef(Typography))
                          className="MuiListItemText-primary"
                          component="span"
                          display="block"
                          variant="body1"
                        >
                          <ForwardRef(Typography)
                            className="MuiListItemText-primary"
                            classes={
                              Object {
                                "alignCenter": "MuiTypography-alignCenter",
                                "alignJustify": "MuiTypography-alignJustify",
                                "alignLeft": "MuiTypography-alignLeft",
                                "alignRight": "MuiTypography-alignRight",
                                "body1": "MuiTypography-body1",
                                "body2": "MuiTypography-body2",
                                "button": "MuiTypography-button",
                                "caption": "MuiTypography-caption",
                                "colorError": "MuiTypography-colorError",
                                "colorInherit": "MuiTypography-colorInherit",
                                "colorPrimary": "MuiTypography-colorPrimary",
                                "colorSecondary": "MuiTypography-colorSecondary",
                                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                "displayBlock": "MuiTypography-displayBlock",
                                "displayInline": "MuiTypography-displayInline",
                                "gutterBottom": "MuiTypography-gutterBottom",
                                "h1": "MuiTypography-h1",
                                "h2": "MuiTypography-h2",
                                "h3": "MuiTypography-h3",
                                "h4": "MuiTypography-h4",
                                "h5": "MuiTypography-h5",
                                "h6": "MuiTypography-h6",
                                "noWrap": "MuiTypography-noWrap",
                                "overline": "MuiTypography-overline",
                                "paragraph": "MuiTypography-paragraph",
                                "root": "MuiTypography-root",
                                "srOnly": "MuiTypography-srOnly",
                                "subtitle1": "MuiTypography-subtitle1",
                                "subtitle2": "MuiTypography-subtitle2",
                              }
                            }
                            component="span"
                            display="block"
                            variant="body1"
                          >
                            <span
                              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                            >
                              <WithStyles(ForwardRef(Typography))
                                variant="srOnly"
                              >
                                <ForwardRef(Typography)
                                  classes={
                                    Object {
                                      "alignCenter": "MuiTypography-alignCenter",
                                      "alignJustify": "MuiTypography-alignJustify",
                                      "alignLeft": "MuiTypography-alignLeft",
                                      "alignRight": "MuiTypography-alignRight",
                                      "body1": "MuiTypography-body1",
                                      "body2": "MuiTypography-body2",
                                      "button": "MuiTypography-button",
                                      "caption": "MuiTypography-caption",
                                      "colorError": "MuiTypography-colorError",
                                      "colorInherit": "MuiTypography-colorInherit",
                                      "colorPrimary": "MuiTypography-colorPrimary",
                                      "colorSecondary": "MuiTypography-colorSecondary",
                                      "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                      "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                      "displayBlock": "MuiTypography-displayBlock",
                                      "displayInline": "MuiTypography-displayInline",
                                      "gutterBottom": "MuiTypography-gutterBottom",
                                      "h1": "MuiTypography-h1",
                                      "h2": "MuiTypography-h2",
                                      "h3": "MuiTypography-h3",
                                      "h4": "MuiTypography-h4",
                                      "h5": "MuiTypography-h5",
                                      "h6": "MuiTypography-h6",
                                      "noWrap": "MuiTypography-noWrap",
                                      "overline": "MuiTypography-overline",
                                      "paragraph": "MuiTypography-paragraph",
                                      "root": "MuiTypography-root",
                                      "srOnly": "MuiTypography-srOnly",
                                      "subtitle1": "MuiTypography-subtitle1",
                                      "subtitle2": "MuiTypography-subtitle2",
                                    }
                                  }
                                  variant="srOnly"
                                >
                                  <span
                                    className="MuiTypography-root MuiTypography-srOnly"
                                  >
                                    Title text Subtitle text
                                  </span>
                                </ForwardRef(Typography)>
                              </WithStyles(ForwardRef(Typography))>
                              <WithStyles(ForwardRef(Typography))
                                aria-hidden={true}
                                variant="body2"
                              >
                                <ForwardRef(Typography)
                                  aria-hidden={true}
                                  classes={
                                    Object {
                                      "alignCenter": "MuiTypography-alignCenter",
                                      "alignJustify": "MuiTypography-alignJustify",
                                      "alignLeft": "MuiTypography-alignLeft",
                                      "alignRight": "MuiTypography-alignRight",
                                      "body1": "MuiTypography-body1",
                                      "body2": "MuiTypography-body2",
                                      "button": "MuiTypography-button",
                                      "caption": "MuiTypography-caption",
                                      "colorError": "MuiTypography-colorError",
                                      "colorInherit": "MuiTypography-colorInherit",
                                      "colorPrimary": "MuiTypography-colorPrimary",
                                      "colorSecondary": "MuiTypography-colorSecondary",
                                      "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                      "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                      "displayBlock": "MuiTypography-displayBlock",
                                      "displayInline": "MuiTypography-displayInline",
                                      "gutterBottom": "MuiTypography-gutterBottom",
                                      "h1": "MuiTypography-h1",
                                      "h2": "MuiTypography-h2",
                                      "h3": "MuiTypography-h3",
                                      "h4": "MuiTypography-h4",
                                      "h5": "MuiTypography-h5",
                                      "h6": "MuiTypography-h6",
                                      "noWrap": "MuiTypography-noWrap",
                                      "overline": "MuiTypography-overline",
                                      "paragraph": "MuiTypography-paragraph",
                                      "root": "MuiTypography-root",
                                      "srOnly": "MuiTypography-srOnly",
                                      "subtitle1": "MuiTypography-subtitle1",
                                      "subtitle2": "MuiTypography-subtitle2",
                                    }
                                  }
                                  variant="body2"
                                >
                                  <p
                                    aria-hidden={true}
                                    className="MuiTypography-root MuiTypography-body2"
                                  >
                                    Title text
                                  </p>
                                </ForwardRef(Typography)>
                              </WithStyles(ForwardRef(Typography))>
                              <WithStyles(ForwardRef(Typography))
                                aria-hidden={true}
                                className="SuggestionItem-subtitle-7"
                                variant="body2"
                              >
                                <ForwardRef(Typography)
                                  aria-hidden={true}
                                  className="SuggestionItem-subtitle-7"
                                  classes={
                                    Object {
                                      "alignCenter": "MuiTypography-alignCenter",
                                      "alignJustify": "MuiTypography-alignJustify",
                                      "alignLeft": "MuiTypography-alignLeft",
                                      "alignRight": "MuiTypography-alignRight",
                                      "body1": "MuiTypography-body1",
                                      "body2": "MuiTypography-body2",
                                      "button": "MuiTypography-button",
                                      "caption": "MuiTypography-caption",
                                      "colorError": "MuiTypography-colorError",
                                      "colorInherit": "MuiTypography-colorInherit",
                                      "colorPrimary": "MuiTypography-colorPrimary",
                                      "colorSecondary": "MuiTypography-colorSecondary",
                                      "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                      "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                      "displayBlock": "MuiTypography-displayBlock",
                                      "displayInline": "MuiTypography-displayInline",
                                      "gutterBottom": "MuiTypography-gutterBottom",
                                      "h1": "MuiTypography-h1",
                                      "h2": "MuiTypography-h2",
                                      "h3": "MuiTypography-h3",
                                      "h4": "MuiTypography-h4",
                                      "h5": "MuiTypography-h5",
                                      "h6": "MuiTypography-h6",
                                      "noWrap": "MuiTypography-noWrap",
                                      "overline": "MuiTypography-overline",
                                      "paragraph": "MuiTypography-paragraph",
                                      "root": "MuiTypography-root",
                                      "srOnly": "MuiTypography-srOnly",
                                      "subtitle1": "MuiTypography-subtitle1",
                                      "subtitle2": "MuiTypography-subtitle2",
                                    }
                                  }
                                  variant="body2"
                                >
                                  <p
                                    aria-hidden={true}
                                    className="MuiTypography-root SuggestionItem-subtitle-7 MuiTypography-body2"
                                  >
                                    Subtitle text
                                  </p>
                                </ForwardRef(Typography)>
                              </WithStyles(ForwardRef(Typography))>
                            </span>
                          </ForwardRef(Typography)>
                        </WithStyles(ForwardRef(Typography))>
                      </div>
                    </ForwardRef(ListItemText)>
                  </WithStyles(ForwardRef(ListItemText))>
                </span>
              </li>
            </ForwardRef(ButtonBase)>
          </WithStyles(ForwardRef(ButtonBase))>
        </ForwardRef(ListItem)>
      </WithStyles(ForwardRef(ListItem))>
    </SuggestionItem>
  </WithStyles(SuggestionItem)>
</injectIntl(WithStyles(SuggestionItem))>
`;
